src/easyscore.ts

```diff
@@ -0,0 +1,492 @@
+      notes.forEach((n: Match) => {
+        if (n) this.addNote(...(n as string[])); // n => [string, string | null, string]
+      });
+    }
+    L('endChord');
+  }
+
+  commitPiece(): void {
+    L('commitPiece');
+    const { factory } = this;
+
+    if (!factory) return;
+
+    const options = { ...this.options, ...this.piece.options };
+
+    // Use the ! operator in the next two lines because reset(...) guarantees that stem and clef are defined.
+    // eslint-disable-next-line
+    const stem: string = options.stem!.toLowerCase(); // e.g., auto | up | down
+    // eslint-disable-next-line
+    const clef: string = options.clef!; // e.g., treble | bass
+
+    const autoStem = stem === 'auto';
+    const stemDirection = !autoStem && stem === 'up' ? StaveNote.STEM_UP : StaveNote.STEM_DOWN;
+
+    // Build StaveNotes.
+    const { chord, duration, dots, type } = this.piece;
+    const keys: string[] = chord.map((note) => note.key + '/' + note.octave);
+    const note = factory.StaveNote({
+      keys,
+      duration,
+      dots,
+      type,
+      clef,
+      auto_stem: autoStem,
+    });
+    if (!autoStem) note.setStemDirection(stemDirection);
+
+    // Attach accidentals.
+    const accids = chord.map((note) => note.accid || null);
+    accids.forEach((accid, i) => {
+      // TODO: Remove "as unknown". This compilation warning will be fixed after factory & accidental are migrated.
+      if (accid) note.addAccidental(i, (factory.Accidental({ type: accid }) as unknown) as Modifier);
+    });
+
+    // Attach dots.
+    for (let i = 0; i < dots; i++) note.addDotToAll();
+
+    this.commitHooks.forEach((fn) => fn(options, note, this));
+
+    this.elements.notes.push(note);
+    this.elements.accidentals.concat((accids as unknown) as Accidental[]); // TODO: FIX THIS!
+    this.resetPiece();
+  }
+}
+
+function setId(options: IDUpdate, note: StaveNote) {
+  if (options.id === undefined) return;
+  note.setAttribute('id', options.id);
+}
+
+function setClass(options: ClassUpdate, note: StaveNote) {
+  if (!options.class) return;
+  const commaSeparatedRegex = /\s*,\s*/;
+  options.class.split(commaSeparatedRegex).forEach((className: string) => note.addClass(className));
+}
+
+export interface EasyScoreOptions {
+  factory?: Factory;
+  builder?: Builder;
+  commitHooks?: CommitHook[];
+  throwOnError?: boolean;
+}
+
+interface EasyScoreDefaults {
+  clef: string;
+  time: string;
+  stem: string;
+  // eslint-disable-next-line
+  [x: string]: any; // allow arbitrary properties via set(defaults)
+}
+
+export class EasyScore {
+  static DEBUG: boolean = false;
+
+  defaults: EasyScoreDefaults;
+  options!: EasyScoreOptions;
+  factory!: Factory;
+  builder!: Builder;
+  grammar!: Grammar;
+  parser!: Parser;
+
+  constructor(options: EasyScoreOptions = {}) {
+    this.setOptions(options);
+    this.defaults = {
+      clef: 'treble',
+      time: '4/4',
+      stem: 'auto',
+    };
+  }
+
+  set(defaults: EasyScoreDefaults): this {
```

Comment:
If `set` is public then `EasyScoreDefaults` should be exported no?

Reply:
Yup. Will export the EasyScoreDefaults interface.

Question!

The next line is `Object.assign(this.defaults, defaults);`

I notice elsewhere in the code base you use Vex.Merge(a, b);

They are very very similar..... the only difference would be that Vex.Merge also copies over properties inherited via the JS prototype chain (e.g., through Object.create()).

We can discuss this later on via a GitHub issue or something, but do you think Vex.Merge can be completely replaced by Object.assign everywhere?

Or are there instances where we depend on copying over properties inherited via prototype chain?? 

I believe that Vex.Merge and Object.assign are equivalent even with properties inherited via es6 `extends` since all properties inherited through `extends` are "own properties" in the instantiated object.

Reply:
`Vex.Merge` is super old. Today I'd rather use the spread operator, something like: `this.defaults = {...this.defaults, ...defaults}`

Reply:
To answer your question specifically, we don't need the properties from the prototype chain. `Object.assign` is almost always better. (However, I feel the spread operator is more readable.)